stemgen/nistemfile.py

```diff
@@ -50,14 +51,17 @@ def _extract_cover(f):
         else:
             data = cover.picture()
         mime = cover.mimeType().lower().strip()
-        if "image/jpeg":
-            fmt = tagpy.mp4.CoverArtFormats.JPEG
-        elif "image/png":
-            fmt = tagpy.mp4.CoverArtFormats.PNG
-        elif "image/bmp":
-            fmt = tagpy.mp4.CoverArtFormats.BMP
-        elif "image/gif":
-            fmt = tagpy.mp4.CoverArtFormats.GIF
+        match mime:
+            case "image/jpeg":
+                fmt = tagpy.mp4.CoverArtFormats.JPEG
+            case "image/png":
+                fmt = tagpy.mp4.CoverArtFormats.PNG
+            case "image/bmp":
+                fmt = tagpy.mp4.CoverArtFormats.BMP
+            case "image/gif":
+                fmt = tagpy.mp4.CoverArtFormats.GIF
+            case _:
+                raise RuntimeError(f"Unsupported cover mime type {mine}")
```

Comment:
I would prefer to return `None` so this will be correctly handled by [the caller](https://github.com/acolombier/stemgen/blob/3a0ad5addf255cb77b327a358b1998eb9f52b1fd/stemgen/nistemfile.py#L172), without the need to having to use exception handling and generic exceptions.

Reply:
That would work, but would now validate the new return type annotation. One possibility would be to annotate that as `Optional`, but looking into `tagpy`, I guess the correct thing to return in the `match` is `tagpy.mp4.CoverArtFormats.Unknown`. I can also add a click warning.

I actually don't expect the `_` case to be ever called, it's just for safety reasons :)

Reply:
>  I guess the correct thing to return in the match is tagpy.mp4.CoverArtFormats.Unknown. I can also add a click warning.

Yeah, I think this is a fair trade-off!